core/src/main/java/org/infinispan/factories/impl/ComponentRegistryTimeTracker.java

```diff
@@ -0,0 +1,236 @@
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
+
+import org.infinispan.commons.time.TimeService;
+import org.infinispan.configuration.global.GlobalConfiguration;
+
+final class ComponentRegistryTimeTracker implements ComponentRegistryTracker {
+
+   private final Map<String, Initialization> components = new ConcurrentHashMap<>(8);
+   private final BasicComponentRegistry registry;
+   private volatile TimeService timeService;
+
+   ComponentRegistryTimeTracker(BasicComponentRegistry registry, boolean global) {
+      this.registry = registry;
+      this.timeService = global ? null : registry.getComponent(TimeService.class).wired();
+   }
+
+   static ComponentRegistryTracker tracker(BasicComponentRegistry registry, boolean global) {
+      return new ComponentRegistryTimeTracker(registry, global);
+   }
+
+   private boolean isTimeServiceAvailable() {
+      return timeService != null;
+   }
+
+   @Override
+   public void instantiating(String componentName) {
+      if (!isTimeServiceAvailable()) return;
+      Initialization initialization = componentInitialization(componentName);
+      initialization.instantiating(timeService.wallClockTime());
+   }
+
+   @Override
+   public void instantiated(String componentName) {
+      if (!isTimeServiceAvailable()) return;
+      Initialization initialization = componentInitialization(componentName);
+      initialization.instantiated(timeService.wallClockTime());
+   }
+
+   @Override
+   public void wiring(String componentName) {
+      if (!isTimeServiceAvailable()) return;
+      Initialization initialization = componentInitialization(componentName);
+      initialization.wiring(timeService.wallClockTime());
+   }
+
+   @Override
+   public void wired(String componentName) {
+      if (!isTimeServiceAvailable()) {
+         if (componentName.equals(GlobalConfiguration.class.getName())) {
+            timeService = registry.getComponent(TimeService.class).wired();
+         }
+         return;
+      }
+      Initialization initialization = componentInitialization(componentName);
+      initialization.wired(timeService.wallClockTime());
+   }
+
+   @Override
+   public void starting(String componentName) {
+      if (!isTimeServiceAvailable()) return;
+      Initialization initialization = componentInitialization(componentName);
+      initialization.starting(timeService.wallClockTime());
+   }
+
+   @Override
+   public void started(String componentName) {
+      if (!isTimeServiceAvailable()) return;
+      Initialization initialization = componentInitialization(componentName);
+      initialization.started(timeService.wallClockTime());
+   }
+
+   @Override
+   public String dump() {
+      StringBuilder sb = new StringBuilder();
+      List<Initialization> c = components.values().stream()
+            .sorted(Comparator.reverseOrder())
+            .toList();
+      for (Initialization initialization : c) {
```

Comment:
I wonder if we should condense the components that took < 1 ms to not spam the log as much as well? Wdyt?